firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,262 @@
+      THEN("The input method reports ok status for all the 255 non-delimeter bytes")
+      THEN("The output method reports waiting status")
+      THEN("The output buffer remains unchanged")
+      THEN("The input method reports output_ready status on the 256th delimeter byte")
+      THEN("After the delimiter is passed as input, The output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: A ChunkSplitter constructed with default parameters correctly handles unacceptably long input chunks
+
+  GIVEN("A uint8_t chunk splitter with a completely full internal buffer of 256 bytes with no delimeters")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("More than 256 non-delimiter bytes are passed as input")
+      THEN("The input method reports invalid_length status on 257th byte and for all the subsequent bytes")
+    
+    WHEN("More than 256 non-delimiter bytes are passed as input, after which a delimiter is passed and output is called")
+      THEN("The input method reports invalid_length status for all the 256 non-delimiter bytes")
+      THEN("The input method reports output ready status on the delimeter byte")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("a non-delimiter byte is passed as input and output is called")
+      THEN("The input method reports invalid_length status")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("The input method reports output_ready status")
+      THEN("the output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: ChunkSplitter constructed with with default value for delimeter and include_delimeter boolean as true correctly handles unacceptably long input chunks
+
+  GIVEN("A uint8_t chunk splitter with a completely full internal buffer of 256 bytes")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("More than 256 non-delimiter bytes are passed as input")
+      THEN("The input method reports invalid_length status on 257th byte and for all the subsequent bytes")
+
+    WHEN("More than 256 non-delimiter bytes are passed as input, after which a delimiter is passed and output is called")
+      THEN("The input method reports invalid_length status for all the 256 non-delimiter bytes")
+      THEN("The input method reports output ready status on the delimeter byte")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
```

Comment:
In each test in this scenario, should I expect the last byte to be a delimiter byte? I think the answer is no